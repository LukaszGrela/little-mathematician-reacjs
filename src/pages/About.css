.about {
    padding-top: 1em;
    text-align: left;
    width: 300px;
    margin: 0 auto;
}
.about .title {
    font-size: 2em;
    line-height: 3em;
}
.about .acknowledgement {
    color: #757575;
}