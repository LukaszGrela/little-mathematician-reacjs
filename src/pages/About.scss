/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

@import '../sass/_variables.scss';
.about {
    padding: 1em;
    text-align: left;
    width: $width-game-column;
    margin: 0 auto;
    .title {
        font-size: 2em;
        line-height: 3em;
    }
    .acknowledgement {
        color: $secondary-text-color;
    }
    .git-source {
        padding: 1rem 0;
        .git-hub-icon {
            vertical-align: bottom;
            width: 1.4rem;
            height: 1.4rem;
            padding-right: 0.4rem;
        }
    }
    
}