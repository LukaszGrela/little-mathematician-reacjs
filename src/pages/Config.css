.config {
    padding-top: 50px;
}
.config label {
    display: block;
    padding-bottom: 8px;
}
.config input,
.config select {
    margin-left: 8px;
}