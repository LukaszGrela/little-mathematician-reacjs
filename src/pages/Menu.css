.menu {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}
.menu button {
    border: none;
    background: none;

    padding: 0;
    margin: 5px;

    width: 70px;
    height: 70px;
}

.menu button svg {
    width: 70px;
    height: 70px;
}

.menu button svg {
    -webkit-transition: all .5s ease;
    transition: all .5s ease;
}

.menu button:hover svg {
    transform:rotate(1turn);
}

.menu button #button-icon-plus {
    fill: #388E3C;
}

.menu button #button-icon-minus {
    fill: #8BC34A;
}

.menu button #button-icon-multiply {
    fill: #CDDC39;
}

.menu button #button-icon-divide {
    fill: #FFEB3B;
}