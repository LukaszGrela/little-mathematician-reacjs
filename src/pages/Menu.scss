/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

@import '../sass/_variables.scss';
.menu {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    @media only screen and (min-device-width: 414px) and (orientation: landscape) {
        transform: translate(-50%, -30%);
    }
    button {
        border: none;
        background: none;
        padding: 0;
        margin: 5px;
        width: 70px;
        height: 70px;
        overflow: visible;
        svg {
            width: 70px;
            height: 70px;
            -webkit-transition: all .5s ease;
            transition: all .5s ease;
        }
        &:hover svg {
            transform: rotate(1turn);
        }
        #button-icon-plus {
            fill: $accent-color-1;
        }
        #button-icon-minus {
            fill: $accent-color-2;
        }
        #button-icon-multiply {
            fill: $accent-color-3;
        }
        #button-icon-divide {
            fill: $accent-color-4;
        }
        .badge {
            position: absolute;
            padding: 4px;
            border-radius: 4px;
            min-width: 20px;
            z-index: 1;
        }
        &.addition .badge {
            background-color: $accent-color-4;
            right: 140px;
            top: -5px;
        }
        &.subtraction .badge {
            background-color: $accent-color-3;
            top: -5px;
            left: 140px;
        }
        &.multiplication .badge {
            background-color: $accent-color-2;
            right: 140px;
            bottom: -5px;
        }
        &.division .badge {
            background-color: $accent-color-1;
            left: 140px;
            bottom: -5px;
        }
    }
}