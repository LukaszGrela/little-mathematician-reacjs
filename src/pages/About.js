/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

import React, { Component } from 'react';

import './About.scss'
import GitHubIcon from '../icons/GitHubIcon';

class About extends Component {
    state = {}
    render() {
        return (
            <div className='about'>
                <div className='title'>Little Mathematician</div>
                <div className='content'>
                    Simple Maths game for children made in ReactJS with aim to explore this framework.
                </div>
                <div className='git-source'>
                    <GitHubIcon/><a href="https://github.com/LukaszGrela/little-mathematician-reacjs" 
                    target='_blank' 
                    rel="noopener noreferrer">LukaszGrela/little-mathematician-reacjs</a>
                </div>
                <div className='acknowledgement'>
                    <ul>
                        <li>Icons in menu (and logo) by Leona Grande from the Noun Project, thenounproject.com/</li>
                        <li>Other icons came from material ui icons, www.materialui.co/</li>
                    </ul>
                </div>
            </div>
        );
    }
}

export default About;
