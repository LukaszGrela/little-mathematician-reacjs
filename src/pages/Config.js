/*
   Copyright 2018 Łukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

import React, { Component } from 'react';

import './Config.css'
import connect from 'react-redux/lib/connect/connect';
import { changeNumberOfQuestions,ConfigType, changeRangeTo, changeRangeFrom } from '../actions/actions';

class Config extends Component {


    constructor(props) {
        super(props);
        this.state = Object.assign({}, props);
    }



    handleChange(field, value) {
        let _val = parseInt(value, 10);
        let gCount = this.props.general.questionCount;
        let gFrom = this.props.general.from;
        let gTo = this.props.general.to;
        const { dispatch } = this.props;
        switch (field) {
            case 'general-qn':
                dispatch(changeNumberOfQuestions(_val, ConfigType.GENERAL));
                break;
            case 'general-rf':
                if (_val >= gTo) {
                    dispatch(changeRangeTo(_val + 10, ConfigType.GENERAL));
                }
                //this.setState({ general: { from: _val, to: gTo, questionCount: gCount } });
                dispatch(changeRangeFrom(_val, ConfigType.GENERAL));
                break;
            case 'general-rt':
                if (_val <= gFrom) {
                    dispatch(changeRangeFrom(_val - 10, ConfigType.GENERAL));
                }
                //this.setState({ general: { to: _val, from: gFrom, questionCount: gCount } });
                dispatch(changeRangeTo(_val, ConfigType.GENERAL));
                break;
            default:
                break;
        }
    }


    render() {
        return (
            <div className='config'>
                <form>
                    <div className='general'>
                        <label>
                            Number of questions:
                            <select value={this.props.general.questionCount}
                                onChange={(e) => this.handleChange('general-qn', e.target.value)}>
                                <option value="10">10</option>
                                <option value="25">25</option>
                                <option value="50">50</option>
                                <option value="100">100</option>
                            </select>
                        </label>
                        <label>
                            Number range from:
                            <select value={this.props.general.from}
                                onChange={(e) => this.handleChange('general-rf', e.target.value)}>
                                <option value="0">0</option>
                                <option value="10">10</option>
                                <option value="20">20</option>
                                <option value="30">30</option>
                                <option value="40">40</option>
                            </select>
                        </label>
                        <label>
                            Number range to:
                            <select value={this.props.general.to}
                                onChange={(e) => this.handleChange('general-rt', e.target.value)}>
                                <option value="10">10</option>
                                <option value="20">20</option>
                                <option value="30">30</option>
                                <option value="40">40</option>
                                <option value="50">50</option>
                            </select>
                        </label>
                    </div>
                    <div className='addition'></div>
                    <div className='subtraction'></div>
                    <div className='multiplication'></div>
                    <div className='division'></div>
                </form>
            </div >
        );
    }
}

const mapStateToProps = (state) => {
    console.log('Config#mapStateToProps', state);
    const config = Object.assign({}, state.config);
    return config;
}

export default connect(mapStateToProps)(Config);