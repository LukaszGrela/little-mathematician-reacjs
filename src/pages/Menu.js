import React, { Component } from 'react';

import './Menu.css'

class Menu extends Component {
    state = {}
    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(action) {
        this.props.onAction && this.props.onAction(action);
    }


    render() {
        return (
            <div className="menu">
                <div className="row">
                    <button onClick={() => {
                        this.handleClick("plus");
                    }}>
                        <img src='./assets/button-icons-plus.svg' />
                    </button>
                    <button onClick={() => {
                        this.handleClick("minus");
                    }}>
                        <img src='./assets/button-icons-minus.svg' />
                    </button>
                </div>
                <div className="row">
                    <button onClick={() => {
                        this.handleClick("multiply");
                    }}>
                        <img src='./assets/button-icons-multiply.svg' />
                    </button>
                    <button onClick={() => {
                        this.handleClick("divide");
                    }}>
                        <img src='./assets/button-icons-divide.svg' />
                    </button>
                </div>
            </div>
        );
    }
}

export default Menu;