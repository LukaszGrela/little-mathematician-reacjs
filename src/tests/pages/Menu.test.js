import React from 'react';
import { shallow } from 'enzyme';
import { Menu } from '../../pages/Menu';
import { GAME_MULTIPLICATION, GAME_ADDITION, GAME_SUBTRACTION, GAME_DIVISION } from '../../gameTypes';

let wrapper,
    onAction,
    element,
    stats = {
        division: '0',
        multiplication: '1',
        subtraction: '2',
        addition: '3',
    };

beforeEach(() => {
    onAction = jest.fn();
    wrapper = shallow(<Menu
        stats={stats}
        onAction={onAction} />);

})

test('Should render Menu correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should onAction be called with GAME_MULTIPLICATION argument on multiplication button click', () => {
    element = wrapper.find('button.multiplication');
    expect(element.length).toBe(1);
    element.simulate('click');
    expect(onAction).toHaveBeenLastCalledWith(GAME_MULTIPLICATION);
});
test('Should onAction be called with GAME_DIVISION argument on division button click', () => {

    element = wrapper.find('button.division');
    expect(element.length).toBe(1);
    element.simulate('click');
    expect(onAction).toHaveBeenLastCalledWith(GAME_DIVISION);
});
test('Should onAction be called with GAME_ADDITION argument on addition button click', () => {

    element = wrapper.find('button.addition');
    expect(element.length).toBe(1);
    element.simulate('click');
    expect(onAction).toHaveBeenLastCalledWith(GAME_ADDITION);
});
test('Should onAction be called with GAME_SUBTRACTION argument on subtraction button click', () => {

    element = wrapper.find('button.subtraction');
    expect(element.length).toBe(1);
    element.simulate('click');
    expect(onAction).toHaveBeenLastCalledWith(GAME_SUBTRACTION);
});

test('Should addition badge display correct value', () => {
    element = wrapper.find('.addition span.badge');
    expect(element.length).toBe(1);
    expect(element.text()).toEqual(stats.addition);
});

test('Should subtraction badge display correct value', () => {
    element = wrapper.find('.subtraction span.badge');
    expect(element.length).toBe(1);
    expect(element.text()).toEqual(stats.subtraction);
});

test('Should multiplication badge display correct value', () => {
    element = wrapper.find('.multiplication span.badge');
    expect(element.length).toBe(1);
    expect(element.text()).toEqual(stats.multiplication);
});

test('Should division badge display correct value', () => {
    element = wrapper.find('.division span.badge');
    expect(element.length).toBe(1);
    expect(element.text()).toEqual(stats.division);
});