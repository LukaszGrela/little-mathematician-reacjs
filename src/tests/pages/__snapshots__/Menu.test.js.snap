// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Menu correctly 1`] = `
<div
  className="menu"
>
  <div
    className="row"
  >
    <button
      className="addition"
      onClick={[Function]}
    >
      <span
        className="badge score"
      >
        3
      </span>
      <ButtonIconDivide />
    </button>
    <button
      className="subtraction"
      onClick={[Function]}
    >
      <span
        className="badge score"
      >
        2
      </span>
      <ButtonIconMinus />
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="multiplication"
      onClick={[Function]}
    >
      <span
        className="badge score"
      >
        1
      </span>
      <ButtonIconMultiply />
    </button>
    <button
      className="division"
      onClick={[Function]}
    >
      <span
        className="badge score"
      >
        0
      </span>
      <ButtonIconDivide />
    </button>
  </div>
</div>
`;
